libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3862,6 +3874,25 @@ void TypeChecker::endVisit(Literal const& _literal)
 					"Literal suffix functions must return exactly one value."
 				);
 			}
+
+			if (ranges::any_of(
+					suffixFunctionType->returnParameterTypes(),
+					[](Type const* returnType) {
+						auto referenceType = dynamic_cast<ReferenceType const*>(returnType);
+						auto mappingType = dynamic_cast<MappingType const*>(returnType);
+						return
+							mappingType ||
+							(referenceType && !referenceType->dataStoredIn(DataLocation::Memory))
+						;
+					}
+				)
+			)
```

Comment:
This is how we'd usually wrap it:
```suggestion
			if (ranges::any_of(
				suffixFunctionType->returnParameterTypes(),
				[](Type const* returnType) {
					auto referenceType = dynamic_cast<ReferenceType const*>(returnType);
					auto mappingType = dynamic_cast<MappingType const*>(returnType);
					return
						mappingType ||
						(referenceType && !referenceType->dataStoredIn(DataLocation::Memory));
				}
			))
```

Reply:
Done.